exchange/exchange.go

```diff
@@ -188,17 +189,46 @@ func (e *exchange) HoldAuction(ctx context.Context, r AuctionRequest, debugLog *
 		_, targData.cacheHost, targData.cachePath = e.cache.GetExtCacheData()
 	}
 
-	responseDebugAllow, accountDebugAllow, debugLog := getDebugInfo(r.BidRequest, requestExt, r.Account.DebugAllow, debugLog)
+	responseDebugAllow, accountDebugAllow, debugLog := getDebugInfo(r.BidRequestWrapper.BidRequest, requestExt, r.Account.DebugAllow, debugLog)
+
+	if r.RequestType == metrics.ReqTypeORTB2Web || r.RequestType == metrics.ReqTypeORTB2App {
+		//Extract First party data for auction endpoint only
+		if responseDebugAllow {
+			//ExtractFPDForBidders removes FPD related structures from request.
+			//In order to preserve original request and return it properly in response.ext.debug.resolvedrequest
+			//original request should be copied before FPD execution
+			resolvedBidReq, err := json.Marshal(r.BidRequestWrapper.BidRequest)
+			if err != nil {
+				return nil, err
+			}
+			r.ResolvedBidRequest = resolvedBidReq
+		}
```

Comment:
Discussed offline. Move `if` statement on line 196 above `if` on line 194 so that the resolved bid request is set in the same way no matter what the request type is.